Packs/Oracle_IAM/Integrations/OracleIAM/OracleIAM.py

```diff
@@ -0,0 +1,752 @@
+import demistomock as demisto
+from CommonServerPython import *
+import traceback
+import base64
+import urllib3
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+ERROR_CODES_TO_SKIP = [
+    404
+]
+
+'''CLIENT CLASS'''
+
+
+def build_body_request_for_update_user(old_user_data, new_user_data):
+    operations = []
+    for key, value in new_user_data.items():
+        operation = {
+            'op': 'replace' if key in old_user_data.keys() else 'add',
+            'path': key,
+            'value': [value] if key in ('emails', 'phoneNumbers') and not isinstance(value, list) else value,
+        }
+        operations.append(operation)
+
+    data = {
+        'schemas': ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+        'Operations': operations,
+    }
+
+    return data
+
+
+class Client(BaseClient):
+    """ A client class that implements logic to authenticate with the application. """
+
+    def __init__(self, base_url, verify=True, proxy=False, ok_codes=tuple(), headers=None, client_id=None,
+                 client_secret=None):
+        super().__init__(base_url, verify, proxy, ok_codes, headers)
+        self.base_url = base_url
+        self.verify = verify
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.headers = headers
+        self.headers['Authorization'] = f'Bearer {self.get_access_token()}'
+
+    def get_access_token(self):
+        client_id_and_secret = f'{self.client_id}:{self.client_secret}'
+
+        # Standard Base64 Encoding
+        encodedBytes = base64.b64encode(client_id_and_secret.encode('utf-8'))
+        encodedStr = str(encodedBytes, 'utf-8')
+
+        headers = {
+            'Authorization': f'Basic {encodedStr}',
+            'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+        }
+
+        data = {
+            'grant_type': 'client_credentials',
+            'scope': 'urn:opc:idm:__myscopes__',
+        }
+
+        try:
+            token = self._http_request('POST', url_suffix='/oauth2/v1/token', headers=headers, data=data)
+        except DemistoException as e:
+            return e
```

Comment:
No need to wrap with try-except

Reply:
I added this to handle better the unauthorized error